Sources/Submissions/AbsentValueStrategy.swift

```diff
@@ -14,6 +16,8 @@ public enum AbsentValueStrategy {
 extension AbsentValueStrategy {
     func valueIfPresent<T: CustomStringConvertible>(_ value: T?) -> T? {
         switch (self, value) {
+        case (.nil, let file as File):
+            return file.data.isEmpty ? nil : value
```

Comment:
smart but feels a little hacky. If we're going to make an exception for File it feels like we should maybe introduce a protocol. The `AbsentValueStrategy` could then be `nil`, `default` (as per protocol), or `equal`. Alternatively we could allow passing in a closure. So adding a case to `AbsentValueStrategy` like: `custom((T?) -> T)` or `custom((T?) -> Bool)`.